File: app/src/main/java/com/example/mobv2/callbacks/CommentCallback.java
package com.example.mobv2.callbacks;

import android.util.Log;

import com.example.mobv2.ui.activities.MainActivity;
import com.example.mobv2.ui.fragments.comments.CommentsFragment;
import com.google.gson.internal.LinkedTreeMap;

import serverapi.MOBServerAPI;

public class CommentCallback implements MOBServerAPI.MOBAPICallback
{
    private final MainActivity mainActivity;
    private final CommentsFragment.Callback callback;

    public CommentCallback(MainActivity mainActivity,
                           CommentsFragment.Callback callback)
    {
        this.mainActivity = mainActivity;
        this.callback = callback;
    }

    @Override
    public void funcOk(LinkedTreeMap<String, Object> obj)
    {
        Log.v("DEBUG", obj.toString());

        var response = (LinkedTreeMap<String, Object>) obj.get("response");

        String commentId = String.valueOf((((Double) response.get("comment_id"))).intValue());

        callback.createCommentByIdAndAddToPosts(commentId);
    }

    @Override
    public void funcBad(LinkedTreeMap<String, Object> obj)
    {
        Log.v("DEBUG", obj.toString());
    }

    @Override
    public void fail(Throwable obj)
    {
        Log.v("DEBUG", obj.toString());
    }
}

File: app/src/main/java/com/example/mobv2/callbacks/CommentCallback.kt
package com.example.mobv2.callbacks

import android.util.Log
import com.example.mobv2.callbacks.abstractions.CommentOkCallback
import com.example.mobv2.ui.activities.MainActivity
import com.example.mobv2.utils.abstractions.responseStateCallbacks.HasOkStateCallback
import serverapi.MOBServerAPI.MOBAPICallback
import com.google.gson.internal.LinkedTreeMap

class CommentCallback(private val mainActivity: MainActivity) : MOBAPICallback, HasOkStateCallback<CommentOkCallback>
{
    private var okCallback: CommentOkCallback? = null

    override fun funcOk(obj: LinkedTreeMap<String, Any>)
    {
        Log.v("DEBUG", obj.toString())
        val response = obj["response"] as LinkedTreeMap<String, Any>?
        val commentId = (response!!["id"] as Double?)!!.toInt().toString()

        okCallback?.createCommentByIdAndAddToPosts(commentId)
    }

    override fun funcBad(obj: LinkedTreeMap<String, Any>)
    {
        Log.v("DEBUG", obj.toString())
    }

    override fun fail(obj: Throwable)
    {
        Log.v("DEBUG", obj.toString())
    }

    override fun setOkCallback(callback: CommentOkCallback)
    {
        this.okCallback = callback;
    }
}